### PR TITLE
fix(help panel): prevent duplicated key displays

### LINKED ISSUE #5037: `Binding.key_display` and help panel interact poorly for comma-separated Binding definitions
If you declare a binding with commas e.g. `Binding("j,down,minus", ..., key_display="foo")`, the key panel will display that same "key display" for each key, resulting in repetition.

I think deduping the key displays might be enough here?

image

### PR DESCRIPTION
**Pull request by TomJGooding**
Fixes #5037

**Please review the following checklist.**

- [ ] Docstrings on all new or modified functions / classes 
- [ ] Updated documentation
- [x] Updated CHANGELOG.md (where appropriate)

### CODE REVIEW

**Comment by willmcgugan** [comment]
I don't think this will de-duplicate the key displays, except where we have identical keys to the same action.

i.e something like this:

```python
BINDINGS = [
    ("foo", "do_foo"),
    ("foo", "do_foo"),
    ("bar", "do_foo")
]
```

Should display `foo bar` for the key display.

**Comment by willmcgugan** [comment]
Consider the following:

```python
from textual.app import App, ComposeResult

from textual.binding import Binding
from textual.widgets import Footer


class HelpPanelApp(App):
    BINDINGS = [
        Binding("a", "bell", "Ring the bell", key_display="bar"),
        Binding("b,e,l", "bell", "Ring the bell", key_display="foo"),
    ]

    def compose(self) -> ComposeResult:
        yield Footer()


HelpPanelApp().run()
```

This will only display a single key, when there are multiple that would apply.

A contrived example perhaps, but even more likely now we have keymaps.

**Comment by TomJGooding** [comment]
Thanks Will for clarifying. I see your point about this edge case, but why would you define multiple bindings like this? The key panel also wouldn't account for these bindings having different descriptions, so why should the key display be different?

**Comment by TomJGooding** [comment]
I _think_ I now understand what you meant by this not de-duping the key displays, now that support for keymaps has been released.

```python
from textual.app import App, ComposeResult
from textual.binding import Binding
from textual.widgets import Footer, Label


class MyApp(App[None]):
    BINDINGS = [
        Binding(
            key="i,up",
            action="increment",
            description="Increment",
            id="app.increment",
            key_display="INCORRECT",
        ),
    ]

    def compose(self) -> ComposeResult:
        yield Label("Check the footer and help panel")
        yield Footer()

    def on_mount(self) -> None:
        self.action_show_help_panel()
        self.set_keymap({"app.increment": "k,plus,j,l"})

    def get_key_display(self, binding: Binding) -> str:
        if binding.id == "app.increment":
            return "correct"
        return super().get_key_display(binding)


if __name__ == "__main__":
    app = MyApp()
    app.run()
```

**Comment by TomJGooding** [comment]
@willmcgugan Friendly ping as currently the duplication of key displays is a bit of a wart in the help panel.

The updated `test_keymap_bindings_key_display` snapshot shows the difference after this fix. But please let me know if I might have still missed something.

File: src/textual/widgets/_key_panel.py [hunk 1]
```diff
@@ -90,11 +90,14 @@ def render_description(binding: Binding) -> Text:
             get_key_display = self.app.get_key_display
             for multi_bindings in action_to_bindings.values():
                 binding, enabled, tooltip = multi_bindings[0]
-                key_display = " ".join(
-                    get_key_display(binding) for binding, _, _ in multi_bindings
-                )
+                unique_key_displays: list[str] = []
```
Comment (willmcgugan) [inline src/textual/widgets/_key_panel.py:93]: I'd be tempted to go with this. Faster, but uses an idiom not everyone is familiar with.

```python
keys_display = " ".join(
    dict.fromkeys(
        get_key_display(binding) for binding, _, _ in multi_bindings
    )
)
```

**Review by willmcgugan — APPROVED** [review]
Thanks. Just a suggestion, which I'm happy to leave as "your call"!

File: src/textual/widgets/_key_panel.py (hunk 1 above)
Comment (TomJGooding) [reply src/textual/widgets/_key_panel.py]: Thanks, I wasn't aware of `dict.fromkeys`.

**Pull request merged**